--- a/src/tools/java/com/google/devtools/build/android/BUILD
+++ b/src/tools/java/com/google/devtools/build/android/BUILD
@@ -75,6 +75,16 @@
     ],
     deps = [
         ":databinding_exec_jar",
+    ],
+)
+
+java_library(
+    name = "gen_base_classes_worker_lib",
+    srcs = ["GenBaseClassesWorker.java"],
+    deps = [
+        ":android_databinding_wrapper_lib",
+        "@bazel_worker_java//:worker_protocol_java_proto",
+        "@bazel_worker_java//src/main/java/com/google/devtools/build/lib/worker:work_request_handlers",
     ],
 )
 

