diff --git a/rules/java.bzl b/rules/java.bzl
--- a/rules/java.bzl
+++ b/rules/java.bzl
@@ -465,7 +465,7 @@

     # Set reasonable max heap default. Required to prevent runaway memory usage.
     # Can still be overridden by callers of this method.
-    jvm_flags = ["-Xms3G", "-Xmx3G", "-XX:+ExitOnOutOfMemoryError"] + jvm_flags
+    jvm_flags = ["-Xms8G", "-Xmx8G", "-XX:+ExitOnOutOfMemoryError"] + jvm_flags

     # executable should be a File or a FilesToRunProvider
     jar = args.get("executable")
