--- a/rules/data_binding.bzl
+++ b/rules/data_binding.bzl
@@ -67,7 +67,7 @@ def _gen_sources(ctx, output_dir, java_package, deps, layout_info, data_binding_
     args.add("-classInfoOut", class_info)
     args.add("-sourceOut", srcjar)
     args.add("-zipSourceOutput", "true")
-    args.add("-useAndroidX", "false")
+    args.add("-useAndroidX", "true")

     if deps:
         if type(deps[0].class_infos) == "depset":
